util/reggen/fpv_csr.sv.tpl

```diff
@@ -36,52 +36,82 @@ module ${block.name}_csr_assert_fpv import tlul_pkg::*; (
 
 <%
   addr_msb = block.addr_width - 1
+  mask = block.addr_width
 %>\
   // declare common read and write sequences
   sequence device_wr_S(logic [${addr_msb}:0] addr);
-    h2d.a_address == addr && h2d.a_opcode inside {PutFullData, PutPartialData} && h2d.a_valid && h2d.d_ready && !d2h.d_valid;
+    h2d.a_address == addr && h2d.a_opcode inside {PutFullData, PutPartialData} &&
+        h2d.a_valid && h2d.d_ready && !d2h.d_valid;
+  endsequence
+
+  sequence device_wc_S(logic [${addr_msb}:0] addr, bit clear_bit, bit [DWidth-1:0] mask);
+    h2d.a_address == addr && h2d.a_opcode inside {PutFullData, PutPartialData} &&
+        h2d.a_valid && h2d.d_ready && !d2h.d_valid && ((h2d.a_data & mask) == clear_bit);
   endsequence
 
   sequence device_rd_S(logic [${addr_msb}:0] addr);
-    h2d.a_address == addr && h2d.a_opcode inside {Get} && h2d.a_valid && h2d.d_ready && !d2h.d_valid;
+    h2d.a_address == addr && h2d.a_opcode inside {Get} && h2d.a_valid && h2d.d_ready &&
+        !d2h.d_valid;
   endsequence
 
   // declare common read and write properties
-  property wr_P(int width, bit [${addr_msb}:0] addr, bit [31:0] compare_data, bit regen = 1);
-    logic [31:0] id;
-    logic [width:0] data;
-    (device_wr_S(addr),id = h2d.a_source, data = h2d.a_data & a_mask_bit) |->
+  property wr_P(bit [${addr_msb}:0] addr, bit [DWidth-1:0] compare_data, bit regen,
+                bit [DWidth-1:0] mask);
+    logic [DWidth-1:0] id, data;
+    (device_wr_S(addr), id = h2d.a_source, data = h2d.a_data & a_mask_bit & mask) |->
+        strong(##[1:$] (d2h.d_valid && d2h.d_source == id && (d2h.d_error ||
+        compare_data == data || !regen)));
+  endproperty
+
+  property wr_ext_P(bit [${addr_msb}:0] addr, bit [DWidth-1:0] compare_data, bit regen,
+                bit [DWidth-1:0] mask);
```

Comment:
nit: alignment of `bit...`  (2 more instances of this below)

Reply:
Thank you for the good catch, updated.